types/uid2/index.d.ts

```diff
@@ -0,0 +1,8 @@
+// Type definitions for uid2 0.3.3
+// Project: https://github.com/coreh/uid2
+// Definitions by: Levi Bostian <https://github.com/levibostian>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export default function(length: number, callback?: Callback): string;
```

Comment:
I would suggest this change:

```suggestion
declare function uid2(length: number): string;
declare function uid2(length: number, callback: (err: Error | null, result?: string) => void): void;
export default uid2;
```

Along with removing the extranous `Callback` type.

Reply:
Thank you for your suggest. I can see a few ways in which your suggestion improves the typings. 

However, I do not understand the reasoning behind using 2 function declarations instead of 1 with the callback parameter as optional? I decided to use 1 function after reading [this part](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#use-optional-parameters) of the do's and don'ts guide in the handbook.

Reply:
The signatures I proposed differ in *more* than just trailing parameters, they also differ in return types. The underlying function at runtime will return a `string` if you do *not* pass a callback, and returns nothing if you *do* pass a callback.

Reply:
Ah, yes. Thank you very much for pointing that out for me.